### ISSUE TITLE
Unable to use multiple worker (num_works) in pyTorch

### ISSUE THREAD

**Issue by rMoerkr** [microsoft/nni#3161]
**Environment**:
- NNI version:1.9
- NNI mode (local|remote|pai):locak
- Client OS: Windows 10 20H2
- Server OS (for remote mode only):
- Python version: 3.8.6 64-bit
- PyTorch/TensorFlow version: 1.7.0+cu110
- Is conda/virtualenv/venv used?: no
- Is running in Docker?: no

**Log message**:
 - nnimanager.log: 
2020-12-6 9:58:17 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:58:18 PM] DEBUG [ 'GET: /version: body:\n{}' ]
[2020-12-6 9:58:19 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:58:19 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:58:19 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:58:19 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:58:19 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:58:19 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:58:22 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:27 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:29 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:58:29 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:58:29 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:58:29 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:58:29 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:58:29 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:58:32 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:37 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:40 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:58:40 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:58:40 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:58:40 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:58:40 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:58:40 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:58:42 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:47 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:51 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:58:51 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:58:51 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:58:51 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:58:51 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:58:51 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:58:52 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:58:57 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:59:02 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:59:02 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:59:02 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:59:02 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:59:02 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:59:02 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:59:02 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:59:06 PM] DEBUG [ 'NNIManager received trial job metrics: [object Object]' ]
[2020-12-6 9:59:06 PM] DEBUG [ 'Sending metrics, job id: V3Peo, metrics: {"parameter_id": 0, "trial_job_id": "V3Peo", "type": "PERIODICAL", "sequence": 0, "value": "0.6897430868797804"}\n' ]
[2020-12-6 9:59:06 PM] DEBUG [ 'ipcInterface command type: [ME], content:[{"parameter_id": 0, "trial_job_id": "V3Peo", "type": "PERIODICAL", "sequence": 0, "value": "0.6897430868797804"}\n]' ]
[2020-12-6 9:59:07 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:59:12 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:59:13 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:59:13 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:59:13 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:59:13 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:59:13 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:59:13 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:59:17 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:59:22 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:59:24 PM] DEBUG [ 'GET: /experiment: body:\n{}' ]
[2020-12-6 9:59:24 PM] DEBUG [ 'GET: /trial-jobs: body:\n{}' ]
[2020-12-6 9:59:24 PM] DEBUG [ 'GET: /metric-data: body:\n{}' ]
[2020-12-6 9:59:24 PM] DEBUG [ 'getTrialJobsByReplayEvents begin' ]
[2020-12-6 9:59:24 PM] DEBUG [ 'getTrialJobsByReplayEvents done' ]
[2020-12-6 9:59:24 PM] DEBUG [ 'GET: /check-status: body:\n{}' ]
[2020-12-6 9:59:27 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
[2020-12-6 9:59:32 PM] DEBUG [ 'ipcInterface command type: [PI], content:[]' ]
 - dispatcher.log:
[12/06/2020, 09:57:57 PM] DEBUG (nni.protocol/Thread-1) Sending command, data: [b'TR00000000000354{"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"batch_size": 128, "lr": 0.001, "momentum": 0.5204947496957271, "pretrained": false, "weight_decay": 0.467437624836745, "CJ_hue": 0.023763575144134463, "CJ_brightness": 0.6117236799388448, "RandomHorizontalFlip": 0.0, "RandomVerticalFlip": 0.5, "TRIAL_BUDGET": 2}, "parameter_index": 0}']
[12/06/2020, 09:58:02 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:02 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:02 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:07 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:07 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:07 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:12 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:12 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:12 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:17 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:17 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:17 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:22 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:22 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:22 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:27 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:27 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:27 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:32 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:32 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:32 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:37 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:37 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:37 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:42 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:42 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:42 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:47 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:47 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:47 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:52 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:52 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:52 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:58:57 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:58:57 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:58:57 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:59:02 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:59:02 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:59:02 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:59:06 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'ME00000000000113']
[12/06/2020, 09:59:06 PM] DEBUG (nni.protocol/MainThread) Received command, data: [{"parameter_id": 0, "trial_job_id": "V3Peo", "type": "PERIODICAL", "sequence": 0, "value": "0.6897430868797804"}
]
[12/06/2020, 09:59:06 PM] DEBUG (nni.msg_dispatcher_base/Thread-2) process_command: command: [CommandType.ReportMetricData], data: [OrderedDict([('parameter_id', 0), ('trial_job_id', 'V3Peo'), ('type', 'PERIODICAL'), ('sequence', 0), ('value', '0.6897430868797804')])]
[12/06/2020, 09:59:06 PM] DEBUG (nni.msg_dispatcher/Thread-2) GOOD
[12/06/2020, 09:59:07 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:59:07 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:59:07 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:59:12 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
[12/06/2020, 09:59:12 PM] DEBUG (nni.protocol/MainThread) Received command, data: []
[12/06/2020, 09:59:12 PM] DEBUG (nni.msg_dispatcher_base/Thread-1) process_command: command: [CommandType.Ping], data: []
[12/06/2020, 09:59:17 PM] DEBUG (nni.protocol/MainThread) Received command, header: [b'PI00000000000000']
 - nnictl stdout and stderr:
 

**What issue meet, what's expected?**:
My code:
```
data_dir = args['data_dir']
       image_datasets = {x: datasets.ImageFolder(os.path.join(data_dir, x),
                                              data_transforms[x])
                      for x in ['train', 'val']}
       dataloaders = {x: torch.utils.data.DataLoader(image_datasets[x], batch_size=args['batch_size'],
                                                     shuffle=True, num_workers=1, pin_memory = True)
                      for x in ['train', 'val']}
       dataset_sizes = {x: len(image_datasets[x]) for x in ['train', 'val']}
       class_names = image_datasets['train'].classes
       device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
------------in train()----------------
 if batch_number % round(args['val_frequency']*(64/args['batch_size'])) == 0: # in batch validation
                        inepoch_val_acc = test_model(model,criterion, dataloaders['val'],device,optimizer)
                        if inepoch_val_acc > best_acc:
                            best_acc = inepoch_val_acc
                            best_model_wts = copy.deepcopy(model.state_dict())
                        nni.report_intermediate_result(inepoch_val_acc)
```
Just ordinary pyTorch mnist demo, but when I turn the num_workers >0, the log of trial become:
> ���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������[12/06/2020, 10:01:32 PM] PRINT inepoch val 60 Loss: 0.5238 Acc: 0.7841
> [12/06/2020, 10:01:42 PM] PRINT minibatch 80/174 Loss: 0.5142 Acc: 0.7910

When I refresh it, sometimes it just has gone and said "No logs available."
And the default metric is a random number that never updates and any connection with my output, like 0.786383 or sth.
The code is running well without NNI framework:
>[12/06/2020, 10:23:59 PM] WARNING (nni) Requesting parameter without NNI framework, returning empty dict
>[12/06/2020, 10:25:33 PM] INFO (nni) Intermediate result: 0.9339086095312806  (Index 0)
>[12/06/2020, 10:27:03 PM] INFO (nni) Intermediate result: 0.917042557364189  (Index 1)

**Comment by scarlett2018** [comment]
This is an known issue, we (@liuzhe-lz) are trying to reproduce the issue locally. @rMoerkr - stay tuned, and we might ping you for more information shortly.

**Comment by kvartet** [comment]
@liuzhe-lz - would you mind sharing the latest status of this issue? thanks :)

**Comment by liuzhe-lz** [comment]
This should have been fixed.
